Jt unit tests

JT421 opened the pull request:
Created the empty container classes as well as the Test Classes for each.

JT421 committed:
Create files

noe-delamora-univar committed:
adding stack class and unittest class for stack

noedlm committed:
adding test classes and barebones container classes

added python unittest link to readme

JT421 committed:
Merge remote-tracking branch 'origin/Noe-Unit-Tests' into JT-Unit-Tests

JT421 committed:
Merge remote-tracking branch 'origin/Noe-Unit-Tests' into JT-Unit-Tests

JT421 committed:
classes

Create empty container classes, create/update test classes for each

JT421 committed:
tabs to spaces

drazax commented on Test_OrderedSet.py:
```diff
@@ -0,0 +1,67 @@
+class TestOrderedSetMethods(unittest.TestMethod):
+    def test_constructor(self):
+        try:
+            orderedset_a = OrderedSet()
```
should split these up because technically if one fails we won't know which one

drazax commented on Test_OrderedSet.py:
```diff
@@ -0,0 +1,67 @@
+    def test_size(self):
+        try:
+            orderedset = OrderedSet()
+            orderedset.size()
```
because size returns something i think the proper thing to do is to test the return value (for clarity, even though you do test size later)

also by testing the return value you test the existence of the function as well

drazax commented on Test_OrderedSet.py:
```diff
@@ -0,0 +1,67 @@
+            print('TEST SIZE: FAIL')
+
+    def test_list_initialization(self):
+        orderedset_a = OrderedSet([1,2,3,4,5])
```
since this is where you're testing size, you should test size 0 too

drazax commented on Test_OrderedSet.py:
```diff
@@ -0,0 +1,67 @@
+import unittest
+from containers.orderedset import OrderedSet
+
```
make sure to add tests for type checking

drazax replied:
add comments to all files

drazax commented on Test_OrderedSet.py:
```diff
@@ -0,0 +1,67 @@
+        orderedset = OrderedSet([3,5,1,2,4])
+        orderedset.remove(2)
+        self.assertEqual(orderedset.size(), 4)
+        self.assertFalse(orderedset.contains(4))
```
should be contains(2)

drazax commented on Test_OrderedSet.py:
```diff
@@ -0,0 +1,67 @@
+        self.assertEqual(orderedset, OrderedSet())
+
+    def test_first_last(self):
+        orderedset = OrderedSet([3,5,1,2,4])
```
should split these tests in two because we won't know which one fails (we will, but like if this were more complicated...)

drazax commented on Test_Queue.py:
```diff
@@ -0,0 +1,56 @@
+    def test_size(self):
+        try:
+            q = Queue()
+            q.size()
```
see comments from ordered set about these (constructor and size)

drazax commented on Test_UnorderedMap.py:
```diff
@@ -0,0 +1,76 @@
+
+class TestUnorderedMapMethods(unittest.TestMethod):
+    def test_constructor(self):
+        try:
```
see comments in ordered set (constructor and size)

JT421 committed:
Corrections

Added header comments to all files, split test cases where appropriate, simplified initial size tests as well as add tests, added type checking tests

JT421 committed:
change Type Checking

Change Type checking from try/except to assertRaises

drazax commented on Test_UnorderedMap.py:
```diff
@@ -10,25 +15,21 @@ def test_constructor(self):
+        u_map = UnorderedMap()
+        u_map.put('a',1)
+        u_map.assertRaises(TypeError, u_map.put(1, 1))
+        u_map.assertRaises(TypeError, u_map.put('b', 'b'))
```
split into two tests because we don't know which test fails if one does

JT421 committed:
split type checking on u_map
